express-get/server.ts

```diff
@@ -0,0 +1,70 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/pagila',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.get('/api/countries', async (req, res, next) => {
+  try {
+    const sql = `
+select
+"countries"."name" as "country-name","countries"."countryId" as "countryId",
+count (*) as "cities"
+from
+"cities"
+join "countries" using ("countryId")
+group by "countries"."name","countries"."countryId"
+
+`;
+    const result = await db.query(sql);
+    const countries = result.rows;
+    if (!countries) {
+      throw new ClientError(404, `country was not found.`);
+    }
+    res.json(countries);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/cities/:cityId', async (req, res, next) => {
+  try {
+    const { cityId } = req.params;
+
+    if (cityId === undefined)
+      throw new ClientError(404, 'the city id was not found.');
+    if (!cityId)
```

Comment:
These two checks don't check to make sure the cityId is an integer. What you need is this:

`if (!Number.isInteger(+cityId))`